File: 682-Baseball-Game.py
class Solution(object):
    def calPoints(self, operations):
        stk = []

        for x in operations:
            if x.isdigit() or (x[0] == '-' and x[1:].isdigit()):
                stk.append(int(x))
            elif x == '+':
                stk.append(stk[-1] + stk[-2])
            elif x == 'C':
                stk.pop()
            elif x == 'D':
                stk.append(stk[-1] * 2)

        return sum(stk)

File: neetcode/682-Baseball-Game.ts
function sum(stack: number[]): number {
  let sum: number = 0
  for (const num of stack) sum += num
  return sum
}

function calPoints(operations: string[]): number {
  const stk: number[] = []

  for (const op of operations) {
    const n = stk.length

    if (op === '+') {
      stk.push(Number(stk[n - 1]) + Number(stk[n - 2]))
    } else if (op === 'D') {
      stk.push(Number(stk[n - 1]) * 2)
    } else if (op === 'C') {
      stk.pop()
    } else {
      stk.push(Number(op))
    }
  }

  return sum(stk)
}
